opencog/reasoning/pln/rules/evaluation-implication-rule.scm

```diff
@@ -0,0 +1,110 @@
+; =============================================================================
+; Evaluation Implication Rule
+;
+; AndLink
+;   EvaluationLink
+;       A
+;       B
+;   ImplicationLink
+;       A
+;       C
+; |-
+; EvaluationLink
+;   C
+;   B
+; -----------------------------------------------------------------------------
+; Based on Chapter 10: Higher Order Extensional Inference
+; This rule handles some cases of PLN higher order logic that don't
+; require quantifiers
+
+; ImplicationLink PredicateNode:A PredicateNode:A
+; can be handled trivially using the existing first-order PLN rules.
+; This is where we handle a special case.
+
+
+(load "formulas.scm")
+
+(define pln-rule-evaluation-implication
+    (BindLink
+        (VariableList
+            (VariableNode "$A")
+            (VariableNode "$B")
+            (VariableNode "$C"))
+        (AndLink
+            (VariableNode "$A")
+            (VariableNode "$B")
+            (VariableNode "$C")
+            (EvaluationLink
+                (VariableNode "$A")
+                (VariableNode "$B"))
+            (ImplicationLink
+                (VariableNode "$A")
+                (VariableNode "$C")))
+        (ExecutionOutputLink
+            (GroundedSchemaNode "scm: pln-formula-evaluation-implication")
+            (ListLink
+                (VariableNode "$A")
+                (VariableNode "$B")
+                (VariableNode "$C")
+                (EvaluationLink
+                    (VariableNode "$A")
+                    (VariableNode "$B"))
+                (ImplicationLink
+                    (VariableNode "$A")
+                    (VariableNode "$C"))
+                (EvaluationLink
+                    (VariableNode "$C")
+                    (VariableNode "$B"))))))
+
+(define (pln-formula-evaluation-implication A B C AB AC CB)
+    (cog-set-tv!
+        CB
+        (stv
+            (simple-deduction-formula 
+                (cog-stv-strength B)
+                (cog-stv-strength A)
+                (cog-stv-strength C)
+                (cog-stv-strength AB)
+                (cog-stv-strength AC))
+            (*
+                0.9
+                (min
+                    (cog-stv-confidence B)
+                    (cog-stv-confidence A)
+                    (cog-stv-confidence C)
+                    (cog-stv-confidence AC)
+                    (* 0.9 (cog-stv-confidence AB)))))))
+
```

Comment:
There should be 3 discount factors right?  

mem2InhFormula has one.
deductionIndependenceBasedFormula has one over the min( )
inh2MemFormula, which is just mem2InhFormula, has one again.